spec.md

```diff
@@ -147,12 +147,20 @@ that contains both context and data).
   * MUST be a non-empty string
 
 ### source
-* Type: Object
-* Description: This describes the software instance that emits the event at
-  runtime (i.e. the producer). It contains sub-properties (listed below)
+* Type: String
+* Description: This property identifies the source of the event as defined
+  by the publishing organization. The value SHOULD be a URI. The source property
+  value MUST uniquely identify the source within the scope set by the "namespace"
+  property. While the value MAY be a interpreted as a URL, the purpose of
+  it to logically identify the source and not to establish the location of the 
+  event source on the network.
 * Constraints:
   * REQUIRED
-  * MUST contain at least one non-empty sub-property.
+* Examples:
```

Comment:
Let me know if I'm getting ahead of myself: I'm curious in practical use for the OSS case whether this will be easy to provide. Oftentimes the service will have a private IP but will want to share a load-balanced name. Do we expect this is just something that software vendors will need to expose as a config file when their software is an event source?

Reply:
Source is URI not a URL. It's an identifier. Many legitimate event sources can't be identified in public network endpoint terms, at all. URI is a universal and rather successful structured identifier concept. What is "the OSS use-case"?